### PR TITLE
Expose annotation stats for a course's assignments

### PR DESCRIPTION
**Pull request by marcospri**
/ GET http://localhost:8001/dashboard/api/course/1826/assignments/stats

```
[
  {
    "id": 350,
    "title": "localhost (make devdata) HTML Assignment",
    "stats": {
      "annotations": 6,
      "replies": 0,
      "last_activity": "2024-05-15T07:22:26.472842"
    }
  },
  {
    "id": 351,
    "title": "Link selection placement text",
    "stats": {
      "annotations": 0,
      "replies": 0,
      "last_activity": null
    }
  },
  {
    "id": 352,
    "title": "",
    "stats": {
      "annotations": 0,
      "replies": 0,
      "last_activity": null
    }
  },
  {
    "id": 353,
    "title": "localhost (make devdata) Groups assignment ",
    "stats": {
      "annotations": 1,
      "replies": 0,
      "last_activity": "2024-05-15T07:23:34.823135"
    }
  }
]
```

## Testing 

- Checkout https://github.com/hypothesis/h/pull/8709 in H
- Launch a bunch of assignments in https://hypothesis.instructure.com/courses/125, note the course ID
- Open http://localhost:8001/dashboard/api/course/XXX/assignments/stats to fetch the stats

### CODE REVIEW

File: tests/unit/lms/services/h_api_test.py [hunk 1]
```diff
@@ -132,27 +132,44 @@ def test_get_annotations(
 
         assert result == expected_result
 
-    def test_get_assignment_stats(self, h_api, http_service):
+    @pytest.mark.parametrize(
```
Comment (marcospri) [inline tests/unit/lms/services/h_api_test.py:135]: This method are not very interesting, just make an HTTP request and return the response. Doing them both in one test.

**Comment by marcospri** [comment]
I'll leave this unmerged to avoid a rebase hell until we land the FE changes.

**Review by acelaya — APPROVED** [review]
Looks good to me

File: lms/models/grouping.py [hunk 2]
```diff
@@ -10,6 +10,9 @@
 from lms.models._mixins import CreatedUpdatedMixin
 from lms.models.json_settings import JSONSettings
 
+if TYPE_CHECKING:
```
[the remark this answers is not in the document]
Comment (marcospri) [reply lms/models/grouping.py:13]: Yes of course, you can do conditional imports, try/except imports... anything.

THe interesting constant here is TYPE_CHECKING which is true only while running the typechecker.

**Pull request merged**